services/frontend-service/src/ui/components/ServiceLane/ServiceLane.tsx

```diff
@@ -104,18 +105,48 @@ const deriveUndeployMessage = (undeploySummary: UndeploySummary | undefined): st
     }
 };
 
-export const ServiceLane: React.FC<{ application: OverviewApplication; hideMinors: boolean }> = (props) => {
+export const ServiceLane: React.FC<{
+    application: OverviewApplication;
+    hideMinors: boolean;
+}> = (props) => {
     const { application, hideMinors } = props;
     const { authHeader } = useAzureAuthSub((auth) => auth);
 
     const appDetails = useAppDetailsForApp(application.name);
+    const componentRef: React.MutableRefObject<any> = React.useRef();
+
     React.useEffect(() => {
-        getAppDetails(application.name, authHeader);
-    }, [application, authHeader]);
+        if (componentRef.current !== null) {
+            const rect = componentRef.current.getBoundingClientRect();
+            if (rect.top >= 0 && rect.bottom <= window.innerHeight) {
+                if (appDetails.appDetailState === AppDetailsState.NOTREQUESTED) {
+                    getAppDetails(application.name, authHeader);
+                }
+            }
+        }
+        const handleScroll = (): void => {
+            if (componentRef.current !== null) {
```

Comment:
please handle the code duplication. This block looks exactly the same as above. You could just call `handleScroll`